projects/plugins/jetpack/modules/subscriptions/subscribe-overlay/class-jetpack-subscribe-overlay.php

```diff
@@ -0,0 +1,229 @@
+		$template->source         = 'plugin';
+		$template->type           = 'wp_template_part';
+		$template->title          = __( 'Jetpack Subscribe overlay', 'jetpack' );
+		$template->status         = 'publish';
+		$template->has_theme_file = false;
+		$template->is_custom      = true;
+		$template->description    = __( 'An overlay that shows up when someone visits your site', 'jetpack' );
+
+		return $template;
+	}
+
+	/**
+	 * Returns the initial content of the Subscribe Overlay template.
+	 * This can then be edited by the user.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_overlay_template_content() {
+		$site_name        = get_bloginfo( 'name' );
+		$site_description = get_bloginfo( 'description' ); // TODO can be empty, how to handle?
+		$skip_to_content  = __( 'Skip to content', 'jetpack' );
+
+		return <<<HTML
+	<!-- wp:group {"layout":{"type":"constrained","contentSize":"400px"}} -->
+	<div class="wp-block-group">
+		<!-- wp:site-logo {"width":90,"shouldSyncIcon":true,"align":"center","className":"is-style-rounded"} /-->
+	
+		<!-- wp:heading {"textAlign":"center","level":1} -->
+		<h1 class="wp-block-heading has-text-align-center">$site_name</h1>
+		<!-- /wp:heading -->
+
+		<!-- wp:paragraph {"align":"center"} -->
+		<p class='has-text-align-center'>$site_description</p>
+		<!-- /wp:paragraph -->
+
+		<!-- wp:jetpack/subscriptions /-->
+
+		<!-- wp:paragraph {"align":"center","className":"jetpack-subscribe-overlay__to-content"} -->
+		<p class="has-text-align-center jetpack-subscribe-overlay__to-content"><a href="#">$skip_to_content</a> ↓</p>
+		<!-- /wp:paragraph -->
+	</div>
+	<!-- /wp:group -->
+HTML;
+	}
+
+	/**
+	 * Enqueues JS to load overlay.
+	 *
+	 * @return void
+	 */
+	public function enqueue_assets() {
+		if ( $this->should_user_see_overlay() ) {
+			wp_enqueue_style( 'subscribe-overlay-css', plugins_url( 'subscribe-overlay.css', __FILE__ ), array(), JETPACK__VERSION );
+			wp_enqueue_script( 'subscribe-overlay-js', plugins_url( 'subscribe-overlay.js', __FILE__ ), array( 'wp-dom-ready' ), JETPACK__VERSION, true );
+		}
+	}
+
+	/**
+	 * Adds overlay with Subscribe Overlay content.
+	 *
+	 * @return void
+	 */
+	public function add_subscribe_overlay_to_frontend() {
+		if ( $this->should_user_see_overlay() ) { ?>
+					<div class="jetpack-subscribe-overlay">
+						<div class="jetpack-subscribe-overlay__close">
+							<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+								<path d="M5.40456 5L19 19M5 19L18.5954 5" stroke="currentColor" stroke-width="1.5"/>
+							</svg>
+						</div>
+						<div class="jetpack-subscribe-overlay__content">
+							<?php block_template_part( self::BLOCK_TEMPLATE_PART_SLUG ); ?>
+						</div>
+					</div>
+			<?php
+		}
+	}
+
+	/**
+	 * Returns true if a site visitor should see
+	 * the Subscribe Overlay.
+	 *
+	 * @return bool
+	 */
+	public function should_user_see_overlay() {
+		// Only show when viewing frontend single post.
```

Comment:
Comment needs tweaking since this isn't the modal for posts.

Reply:
fixed